src/04-date-tasks.js

```diff
@@ -73,8 +82,31 @@ function isLeapYear(/* date */) {
  *    Date(2000,1,1,10,0,0),  Date(2000,1,1,10,0,0,250)     => "00:00:00.250"
  *    Date(2000,1,1,10,0,0),  Date(2000,1,1,15,20,10,453)   => "05:20:10.453"
  */
-function timeSpanToString(/* startDate, endDate */) {
-  throw new Error('Not implemented');
+function timeSpanToString(startDate, endDate) {
+  function prepareForOutput() {
+    Object.entries(this).forEach(([key, value]) => {
+      if (value.toString().length === 1
+        || (value.toString().length === 2 && key.length === 3)) {
+        if (key.length === 2) {
```

Comment:
Nested "if" makes logic complex and decreases readability.
In this case I would recommend to move this operation in the separate function
```suggestion
const formateDateValue = (value, length) => length === 2 ? `0${value}` :  `00${value}`
```